src/igz_controls/components/default-dropdown/default-dropdown.component.js

```diff
@@ -287,13 +295,28 @@
          */
         function onChangeTypingInput() {
             if (!lodash.isNil(ctrl.typedValue)) {
-                var newItem = {
-                    id: ctrl.typedValue,
-                    visible: true
-                };
-                lodash.set(newItem, ctrl.nameKey || 'name', ctrl.typedValue);
+                if (ctrl.autocomplete) {
+                    ctrl.isDropdownContainerShown = false;
+
+                    ctrl.valuesArray = lodash.filter(valuesArrayCopy, function (item) {
+                        var itemName = ctrl.autocompleteIgnoreCase ? item.name.toLowerCase() : item.name;
+                        var typedValue = ctrl.autocompleteIgnoreCase ? ctrl.typedValue.toLowerCase() : ctrl.typedValue;
```

Comment:
This line can be move to before the filter loop.
There's no reason to re-calculate `typedValue` on each iteration of the filter loop.